docs/articles/documentation/guides/advanced-guides/authentication.md

```diff
@@ -1,50 +1,37 @@
-permalink: /documentation/test-api/authentication/user-roles.html
+title: Authentication
+permalink: /documentation/guides/advanced-guides/authentication.html
 ---
-# User Roles
+# Authentication
+
+TestCafe ships with the [user roles](#user-roles) mechanism that covers most authentication types that require user actions on the page.
+
+You can also test websites that use [HTTP Basic](https://en.wikipedia.org/wiki/Basic_access_authentication) and [NTLM](https://en.wikipedia.org/wiki/NT_LAN_Manager) authentication. See [HTTP/NTLM Authentication](#http-authentication) for more information on how to provide the credentials.
+
+## User Roles
 
 Many test scenarios involve the activity of more than one user. TestCafe addresses these scenarios by providing a convenient way
 to isolate authentication test actions and apply them easily whenever you need to switch the user account.
 
 A piece of logic that logs in a particular user is called a *role*. Define a role for each user account participating in your test.
 
-* [Why Use Roles](#why-use-roles)
-* [Create and Apply Roles](#create-and-apply-roles)
-* [Anonymous Role](#anonymous-role)
-* [Role Options](#role-options)
-  * [options.preserveUrl](#optionspreserveurl)
-* [Troubleshooting](#troubleshooting)
-  * [Test Actions Fail After Authentication](#test-actions-fail-after-authentication)
-
-## Why Use Roles
+### Why Use Roles
 
 Unlike other methods used to [extract reusable test logic](../../recipes/extract-reusable-test-code/README.md), roles were specially designed for login operations and provide the following dedicated features:
 
 * **Object-based API.** Authentication data and logic are stored in an object that is easy to pass around and activate when needed.
-* **Single login.** Login actions are not repeated when you switch to a previously used role within the same session. If you activate a role in the [beforeEach](../test-code-structure.md#test-hooks) hook, login actions run once before the first test. The subsequent tests reuse authentication data so that it happens instantly.
+* **Single login.** Login actions are not repeated when you switch to a previously used role within the same session. If you activate a role in the [beforeEach](../basic-guides/test-organization.md#test-hooks) hook, login actions run once before the first test. The subsequent tests reuse authentication data so that it happens instantly.
 * **Automatic return.** The browser automatically navigates back to the page that was opened when you switched the role (you can disable this behavior if it doesn't suit your scenario).
 * **No logout needed.** Authentication data is automatically cleared when you switch between roles.
 * **Multiple authentication support.** If you log in to different services/websites during a test, authentication data from cookies and browser storages is accumulated in the active role. When you switch back to this role within the same test, you are automatically logged in to all the websites.
```

Comment:
"If you log in to different services/websites during a test, authentication data from cookies and browser storages is accumulated in the active role."
Do you mean:
If you log in to different services/websites during a test, authentication data from cookie and browser storage is saved in the active role.

Reply:
Yes, it's saved, I just wanted to emphasize that after each login, authentication data is added to the same role where data from previous logins is stored. That's why 'accumulated'. But it can be 'saved', that's fine.

Reply:
I understand now. Let's say:
If you log in to different services/websites during a test, authentication data from cookie and browser storage accumulates in the active role.